Gibe.Umbraco.Blog/UnpagedBlogService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Gibe.Umbraco.Blog.Filters;
+using Gibe.Umbraco.Blog.Models;
+using Gibe.Umbraco.Blog.Sort;
+
+namespace Gibe.Umbraco.Blog
+{
+	public class UnpagedBlogService<T> : IUnpagedBlogService<T> where T : class, IBlogPostModel
+	{
+		private readonly IBlogSearch _blogSearch;
+		private readonly IBlogPostMapper<T> _blogPostMapper;
+
+		public UnpagedBlogService(IBlogSearch blogSearch, IBlogPostMapper<T> blogPostMapper)
+		{
+			_blogSearch = blogSearch;
+			_blogPostMapper = blogPostMapper;
+		}
+
+		public UnpagedBlogSearchResults<T> GetPosts(int startPost, int postCount, ISort sort = null,
+			IEnumerable<IBlogPostFilter> filters = null)
+		{
+			if (sort == null) sort = new DateSort();
+			if (filters == null) filters = Enumerable.Empty<IBlogPostFilter>();
+
+			var results = _blogSearch.Search(filters, sort);
+
+			return new UnpagedBlogSearchResults<T>
+			{
+				BlogPosts = _blogPostMapper.ToBlogPosts(results.Skip(startPost != 0 ? startPost - 1 : 0).Take(postCount)),
```

Comment:
Don't think we need the logic in the skip

Reply:
i think this was to prevent error if 0 was passed in but as we discussed, it can still be misused and cause errors in other ways. so yup this logic will be removed.